server/scsynth/SC_CoreAudio.cpp

```diff
@@ -339,8 +347,26 @@ SC_AudioDriver::~SC_AudioDriver() {
 }
 
 void SC_AudioDriver::RunThread() {
-    /* NB: on macOS we just keep the default thread priority */
-#ifdef NOVA_TT_PRIORITY_RT
+#ifdef __APPLE__
+    // on macOS, set the thread priority to user-interactive
+    int result = pthread_set_qos_class_self_np(QOS_CLASS_USER_INTERACTIVE, 0);
+    if (result != 0) {
+        scprintf("SC_AudioDriver: could not set thread priority to user-interactive: %s\n", strerror(result));
+    }
+#    ifdef NOVA_TT_PRIORITY_PERIOD_COMPUTATION_CONSTRAINT
+    int blockSize = mWorld->mBufLength;
+    uint32_t ns_per_block = 1e9 / mSampleRate * blockSize;
+    uint32_t computation = ns_per_block * 0.4;
+    uint32_t constraint = ns_per_block * 0.8;
+    // printf("Verifying values: mSampleRate = %f, blockSize = %d, period = %d, computation = %d, constraint = %d\n",
+    // mSampleRate, blockSize, ns_per_block, computation, constraint);
+
+    bool success = nova::thread_set_priority_rt(ns_per_block, computation, constraint, true);
```

Comment:
As this is the NRT thread, we want a thread priority that is high, but still below the audio thread. This is what `int priority = nova::thread_priority_interval_rt().first;` attempts to do on Linux and Windows. We need to do a similar thing on macOS. AFAICT, your code essentially sets the same priority as the audio thread, which doesn't seem right.

Reply:
Thanks! Where's the audio thread? What's its priority?

This implementation definitely improves scsynth on macOS (less dropouts / more headroom) - I'm not saying it's right but I'm reinforcing that it did _something_.